pkg/sanity/controller.go

```diff
@@ -114,6 +114,7 @@ var _ = DescribeSanity("Controller Service [Controller Server]", func(sc *TestCo
 			ControllerClient: csi.NewControllerClient(sc.ControllerConn),
 			NodeClient:       csi.NewNodeClient(sc.Conn),
 		}
+		r.ControllerPublishSupported = isControllerCapabilitySupported(r, csi.ControllerServiceCapability_RPC_PUBLISH_UNPUBLISH_VOLUME)
```

Comment:
If `r.ControllerPublishSupported` was always false (= unset), why was `ControllerPublishVolume` called (as mentioned in https://github.com/kubernetes-csi/csi-test/issues/316)?

Are there perhaps `ControllerPublishVolume` calls which need to be guarded with an if check?

Reply:
Every test that uses `ControllerPublishVolume` has its [own guard](https://github.com/kubernetes-csi/csi-test/blob/b486fd25346ea68b3c365e96fb7241df8adf64dd/pkg/sanity/controller.go#L1059-L1062)..

If only `ControllerPublishVolume` [can set the](https://github.com/kubernetes-csi/csi-test/blob/b486fd25346ea68b3c365e96fb7241df8adf64dd/pkg/sanity/resources.go#L151) `volumeInfo`'s `NodeId` so maybe [that cl.ControllerPublishSupported check](https://github.com/kubernetes-csi/csi-test/blob/da7a71c90b602694a44815fc4c3533e803601b87/pkg/sanity/resources.go#L295) is redundant and we should just leave the `info.NodeID != ""` there (and remove `ControllerPublishSupported` in `Resources` I added), but I don't know If we wanna just removed guards.

Reply:
I prefer to remove `ControllerPublishSupported`. The code becomes simpler without it and we don't need it because when `info.NodeID != ""`, then we know that `ControllerUnpublishVolume` can and must be called.

Reply:
Let's also remove `NodeStageSupported`. It's only checked once, which can replaced with a `isNodeCapabilitySupported` call.

Reply:
ok, done.